### PR TITLE
Accept java.io.File for filepaths

### LINKED ISSUE #985: `:crux.kv/db-dir` and friends should accept `File` as well as `String`
see config keys that accept `Duration`

### PR DESCRIPTION
**Pull request by danmason**
Resolves #985, also replacing a few usages in the codebase of now unnecessary conversions to string (and leaving some untouched, to check both work fine).

### CODE REVIEW

**Comment by danmason** [comment]
CircleCI build here - not sure why it doesn't appear to have linked to the PR: https://app.circleci.com/pipelines/github/danmason/crux/815/workflows/e45348f4-fcba-4618-bf1b-e8490cf7bbbe

**Review by jarohen — CHANGES_REQUESTED** [review]
Thanks @danmason - could this return `File`s rather than strings, as the `Duration` example does? Cheers

**Comment by jarohen** [comment]
Thinking about it, maybe `Path` might be better, rather than Java's old `File`.

**Comment by jarohen** [comment]
Looks to have linked correctly to me? Strange.

File: crux-lmdb/src/crux/kv/lmdb.clj [hunk 1]
```diff
@@ -80,8 +82,10 @@
 
 (defn- env-open [^long env dir ^long flags]
   (success? (LMDB/mdb_env_open env
-                               (.getAbsolutePath (doto (io/file dir)
-                                                   (.mkdirs)))
+                               (->> (make-array FileAttribute 0)
```
Comment (jarohen) [inline crux-lmdb/src/crux/kv/lmdb.clj]: memo to self: threading use

File: crux-kafka-embedded/src/crux/kafka/embedded.clj [hunk 2]
```diff
@@ -92,18 +95,18 @@
...
-                (start-kafka-broker (merge broker-config
-                                           {"log.dir" (str (io/file kafka-log-dir))
-                                            "port" (str kafka-port)
-                                            "zookeeper.connect" (str *host* ":" zookeeper-port)}))
-                (catch Throwable t
-                  (stop-zookeeper zookeeper)
-                  (throw t)))]
-    (->EmbeddedKafka zookeeper kafka (assoc options :bootstrap-servers (str *host* ":" kafka-port)))))
+  [options]
+  (let [{:crux.kafka.embedded/keys [zookeeper-data-dir zookeeper-port kafka-log-dir kafka-port broker-config]
```
Comment (jarohen) [inline crux-kafka-embedded/src/crux/kafka/embedded.clj:99]: memo to self: conform?

**Review by danmason — COMMENTED** [review]
Note: Getting some test failures, seemingly related to memdb.

**Edit**: Seems to be related to this line, likely also needs updating to be brung in-line with everything else - https://github.com/juxt/crux/blob/master/crux-core/src/crux/kv/memdb.clj#L110

**Pull request merged**